sdk/python/feast/infra/feature_servers/aws_lambda/app.py

```diff
@@ -0,0 +1,23 @@
+from pydantic import StrictBool, StrictStr
+from pydantic.typing import Literal
+
+from feast.repo_config import FeastConfigBaseModel
+
+
+class AwsLambdaFeatureServerConfig(FeastConfigBaseModel):
+    """Feature server config for AWS Lambda."""
+
+    type: Literal["aws_lambda"] = "aws_lambda"
+    """Feature server type selector."""
+
+    enabled: StrictBool
+    """Whether the feature server should be launched."""
+
+    public: StrictBool
+    """Whether the endpoint should be publicly accessible."""
+
+    auth: StrictStr
```

Comment:
Let's set the default to `none` and change the type to be one of `none` and `api-key`
```suggestion
    auth: Literal["none", "api-key"] = "none"
```